### PR TITLE
NVIDIA: Completion fixes

### PR DESCRIPTION
**Pull request by raspawar**
#NVIDIA llm class completion fixes
Added support for nvidia/mistral-nemo-minitron-8b-base model as well

Fixes # (Test Cases, Streaming)

Following up on: https://github.com/run-llama/llama_index/pull/15815
Removing _stream_complete() 
Also, fixed test cases for async

## Type of Change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

cc: @sumitkbh @mattf @dglogo

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_text-completion.py [hunk 1]
```diff
@@ -31,7 +30,8 @@ def __exit__(self, *exc: object) -> None:
             os.environ["NVIDIA_API_KEY"] = self.api_env_was
 
 
-def mock_completion_v1(model_name: str) -> Completion:
+def mock_completion_v1(*args, **kwargs) -> Completion:
+    model_name = kwargs.get("model")
```
Comment (mattf) [inline llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_text-completion.py:34]: the model name doesn't appear to be important here. can you simply make it a static string?

File: llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_text-completion.py [hunk 2]
```diff
@@ -52,9 +52,8 @@ async def mock_async_completion_v1(*args: Any, **kwargs: Any) -> Completion:
     return mock_completion_v1(*args, **kwargs)
 
 
-def mock_chat_completion_stream_v1(
-    model_name: str,
-) -> Generator[Completion, None, None]:
+def mock_completion_stream_v1(*args, **kwargs) -> Generator[Completion, None, None]:
+    model_name = kwargs.get("model")
```
Comment (mattf) [inline llama-index-integrations/llms/llama-index-llms-nvidia/tests/test_text-completion.py:56]: the model name doesn't appear to be important here. can you simply make it a static string?

**Review by mattf — COMMENTED** [review]
the async tests are skipped by default, add the pytest-asyncio dep to enable them: `poetry add --group dev 'pytest-asyncio=^0.23.0'`

does this work with llama-index-llms-openai==0.2.1? if not, update llama-index-llms-openai dep to ^0.2.2

**Comment by mattf** [comment]
you can incorporate https://github.com/run-llama/llama_index/pull/15826 to address the structured output and tools async tests

**Comment by logan-markewich** [comment]
Is this good to merge? If so, plz bump the version

**Comment by raspawar** [comment]
yes it is!

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**